specifyweb/frontend/js_src/lib/components/DataModel/collection.ts

```diff
@@ -104,7 +104,8 @@ function mapValue(
   if (key === 'orderBy') return (value as string).toString().toLowerCase();
   else if (key === 'domainFilter') {
     const scopingField = schema.models[tableName].getScopingRelationship();
-    return value === true && typeof scopingField === 'object'
+    return value === true &&
+      (tableName === 'Attachment' || typeof scopingField === 'object')
```

Comment:
could this problem be better solved by using .getScope() instead of .getScopingRelationship()?
it seems to better resolve scoping for many (though not all - i.e storageattachment) attachment tables - https://github.com/specify/specify7/blob/issue-4307/specifyweb%2Ffrontend%2Fjs_src%2Flib%2Fcomponents%2FDataModel%2F__tests__%2F__snapshots__%2FspecifyTable.test.ts.snap#L1248C10-L1436

Reply:
That would most likely work if Specify were getting the attachment counts on the `table` attachment tables, however the the collection fetch which does so is using the `Attachment` table only, and `scema.models.Attachment.getScope()` returns undefined. 

https://github.com/specify/specify7/blob/332069df197d376ed2f82b557c8598e66d12cd04/specifyweb/frontend/js_src/lib/components/Attachments/index.tsx#L105-L118

Although the suggestion would still better handle other cases in the future, so for now I will include it with this 'fix'.

Reply:
See https://github.com/specify/specify7/pull/3304/commits/97bf9a80747a5cc4b1bbbe2ce8a082e88484cdce